app/src/main/java/app/pwhs/blockads/service/RootProxyService.kt

```
@@ -208,6 +213,19 @@ class RootProxyService : Service() {
                     firewallManager = null
                 }
 
+                // Resolve whitelisted apps to UIDs so iptables skips their
+                // DNS — Root-mode equivalent of VPN mode's
+                // addDisallowedApplication. Without this the whitelist had
+                // no effect at all in Root Proxy mode (#150).
+                whitelistedUids = appPrefs.getWhitelistedAppsSnapshot().mapNotNull { pkg ->
+                    try {
+                        packageManager.getApplicationInfo(pkg, 0).uid
+                    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)